[DOCS-14229] Update Forms documentation (#36493)
DataDog/documentation: content/en/actions/forms/_index.md

---

## Overview

Datadog Forms allow you to collect input, analyze responses, and trigger automations in Datadog. Forms and their responses can be shared across your organization, allowing you to collect and analyze data with your team.

Some ways you can use forms:
- Scaffold services from predefined templates.
- Survey engineering feedback in an internal developer portal (IDP).
- Create service requests and [cases][1] for security, platform, or IT teams directly from employee form responses.

## Create a form

When creating a form, you can use a blueprint, [import an existing form](#import-a-form), or start from scratch. Blueprints are starter forms that cover common use cases. They come loaded with a sample description and questions to help familiarize yourself with form elements, and some blueprints include a pre-configured automation. Blueprints include Developer Experience Survey, IDP Feedback, Case Management Service Request, Report an Incident, Bug Report, On-Call Escalation, Post-Incident Review, and more.

To create a form:
1. On the [Forms][2] page, click {{< ui >}}New Form{{< /ui >}}.

> 1. Select a blueprint, import a form, or start with a blank form, then click {{< ui >}}Continue{{< /ui >}}.

Reviewer: ```suggestion
1. Select a blueprint, import a form, or start with a blank form. Click {{< ui >}}Continue{{< /ui >}}.
```

1. Optionally, name your form and give it a description. Click {{< ui >}}Continue{{< /ui >}}.

> 1. To add a new component, click **Add Component**, or on the Fields side panel, click the **+** icon. See [Form components][3] for the full list of component types and their options.

Reviewer: ```suggestion
1. To add a new component, click **Add Component**, or on the {{< ui >}}Fields{{< /ui >}} side panel, click the plus **+** icon. See [Form components][3] for the full list of component types and their options.
```

1. Click {{< ui >}}Share{{< /ui >}} to copy the form link or configure sharing options.

## Import a form

You can import an existing form into Datadog from a PDF or JSON file.

To import a form:
1. On the [Forms][2] page, click {{< ui >}}New Form{{< /ui >}}.
1. Click {{< ui >}}Import a Form{{< /ui >}}.
1. Choose a source to import your form fields, then follow the prompts.

## Form settings

From the [Forms][2] page, click a form. In the {{< ui >}}Fields{{< /ui >}} panel, click the gear <i class="icon-cog-2"></i> icon to access the following settings:

| Setting | Description |
|---------|-------------|
| Accepting Responses | Toggle to set the form as active or inactive. When inactive, the form does not accept new responses. You can also set an end date to automatically close the form on a specific date. |
| Anonymous Responses | When enabled, respondent emails are not stored. |
| Single Responses | When enabled, prevents the same person from submitting more than one response by tracking user identity. You can enable this setting together with Anonymous Responses. |

> | Manage Permissions | Configure who can view the form and submitted responses. See [Manage access](#manage-access). |

Reviewer: ```suggestion
| Manage Permissions | Configure who can view the form and submit responses. See [Manage access](#manage-access). |
```

Author: Great catch!

title: Forms
description: Build forms to collect input, analyze responses, and trigger automations.
disable_toc: false
further_reading:
  - link: https://www.datadoghq.com/blog/datadog-forms
    tag: Blog
    text: Turn feedback into action across your engineering org with Datadog Forms
  - link: https://www.datadoghq.com/blog/datadog-forms-sheets-developer-feedback/
    tag: Blog
    text: Turn developer feedback into operational insight with Datadog Forms and Sheets

---

## Overview

Datadog Forms allow you to collect input, analyze responses, and trigger automations in Datadog. Forms and their responses can be shared across your organization, allowing you to collect and analyze data with your team.

Some ways you can use forms:
- Scaffold services from predefined templates.
- Collect engineering feedback in an internal developer portal (IDP).
- Create service requests for security, platform, or IT teams directly from employee form responses.

## Create a form

> When creating a form, you can use a blueprint or start from scratch. Blueprints are starter forms that cover common use cases. They come loaded with a sample description and questions to help familiarize yourself with form elements, and some blueprints include a pre-configured automation. Blueprints include Developer Experience Survey, IDP Feedback, Case Management Service Request, Report an Incident, Bug Report, On-Call Escalation, Post-Incident Review, and more.

Author: ```suggestion
When creating a form, you can use a blueprint, [import an existing form](#import-a-form), or start from scratch. Blueprints are starter forms that cover common use cases. They come loaded with a sample description and questions to help familiarize yourself with form elements, and some blueprints include a pre-configured automation. Blueprints include Developer Experience Survey, IDP Feedback, Case Management Service Request, Report an Incident, Bug Report, On-Call Escalation, Post-Incident Review, and more.
```

To create a form:
1. On the [Forms][1] page, click {{< ui >}}New Form{{< /ui >}}.
1. Select a blueprint or start with a blank form, then click {{< ui >}}Continue{{< /ui >}}.
1. Optionally, name your form and give it a description. Click {{< ui >}}Continue{{< /ui >}}.
1. To add a new component, click **Add Component**, or on the Fields side panel, click the **+** icon. See [Form components][2] for the full list of component types and their options.
1. Click {{< ui >}}Save{{< /ui >}}.

> To preview, share, and debug your form:

Author: ```suggestion
To preview and share your form:
```

title: Forms
description: Build forms to collect input, analyze responses, and trigger automations.
disable_toc: false

> further_reading:

Reviewer 2: There is another blog we just posed, can you link it as well? https://www.datadoghq.com/blog/datadog-forms-sheets-developer-feedback/

To create a form:
1. On the [Forms][1] page, click {{< ui >}}New Form{{< /ui >}}.
1. Select a blueprint or start with a blank form, then click {{< ui >}}Continue{{< /ui >}}.
1. Optionally, name your form and give it a description. Click {{< ui >}}Continue{{< /ui >}}.
1. To add a new component, click **Add Component**, or on the Fields side panel, click the **+** icon. See [Form components][2] for the full list of component types and their options.
1. Click {{< ui >}}Save{{< /ui >}}.

To preview, share, and debug your form:
1. Click {{< ui >}}View{{< /ui >}} to display the form as it appears to respondents. Click {{< ui >}}Edit{{< /ui >}} to return to the creator view.
1. Click {{< ui >}}Share{{< /ui >}} to copy the form link or configure sharing options.
1. In the {{< ui >}}Debug{{< /ui >}} section, you can:
    - Click {{< ui >}}Form{{< /ui >}} to show a mini-preview of your form.
    - Click {{< ui >}}Data Definition{{< /ui >}} to show the JSON version of the components' definition.
    - Click {{< ui >}}UI Definition{{< /ui >}} to show the JSON version of the form's UI.

## Import a form

> You can import an existing form into Datadog from a JSON file or migrate directly from Google Forms.

Author: ```suggestion
You can import an existing form into Datadog from a PDF or JSON file.
```

> 

Reviewer 2: Forms are now open to everyone, can you remove the sign up form for preview access? 

<img width="957" height="204" alt="Image" src="https://github.com/user-attachments/assets/682594e2-9125-41ed-a2a9-a305f93c5408" />

Author: Looks like this was already removed!

Datadog Forms allow you to collect input, analyze responses, and trigger automations in Datadog. Forms and their responses can be shared across your organization, allowing you to collect and analyze data with your team.

Some ways you can use forms:
- Scaffold services from predefined templates.
- Collect engineering feedback in an internal developer portal (IDP).

> - Create service requests for security, platform, or IT teams directly from employee form responses.

Reviewer 2: Can we include "Case Management" here and link to Datadog Case Management product?

Author: ```suggestion
- Create service requests and [cases][8] for security, platform, or IT teams directly from employee form responses.
```

- Scaffold services from predefined templates.
- Collect engineering feedback in an internal developer portal (IDP).
- Create service requests for security, platform, or IT teams, directly from employee form responses. 
- Create service requests for security, platform, or IT teams directly from employee form responses.

## Create a form

> When creating a form, you can use a template or start from scratch. Templates are starter forms that cover common use cases. They come loaded with a sample description and questions to help familiarize yourself with form elements. Templates also showcase best practices for setting up form elements. 

Reviewer 2: Today we also have the option to import forms (From PDF, JSON, or Google Forms) can we add it?

Author: This has a separate section under Import a Form, but I added the PDF option there

> 1. Select a blueprint or start with a blank form, then click {{< ui >}}Continue{{< /ui >}}.

Reviewer 2: Same comment here

1. Optionally, name your form and give it a description. Click {{< ui >}}Continue{{< /ui >}}.
1. To add a new component, click **Add Component**, or on the Fields side panel, click the **+** icon. See [Form components][2] for the full list of component types and their options.
1. Click {{< ui >}}Save{{< /ui >}}.

To preview, share, and debug your form:
1. Click {{< ui >}}View{{< /ui >}} to display the form as it appears to respondents. Click {{< ui >}}Edit{{< /ui >}} to return to the creator view.
1. Click {{< ui >}}Share{{< /ui >}} to copy the form link or configure sharing options.

> 1. In the {{< ui >}}Debug{{< /ui >}} section, you can:

Author: @barakshush I couldn't find this, are these still relevant instructions?

Reviewer 2: No

    - Click {{< ui >}}UI Definition{{< /ui >}} to show the JSON version of the form's UI.

## Import a form

You can import an existing form into Datadog from a JSON file or migrate directly from Google Forms.

To import a form:
1. On the [Forms][1] page, click {{< ui >}}New Form{{< /ui >}}.
1. Click {{< ui >}}Import a Form{{< /ui >}}.
1. Choose a source to import your form fields, then follow the prompts.

## Form settings

From the [Forms][1] page, click a form. In the {{< ui >}}Fields{{< /ui >}} panel, click the gear <i class="icon-cog-2"></i> icon to access the following settings:

| Setting | Description |
|---------|-------------|
| Accepting Responses | Toggle to set the form as active or inactive. When inactive, the form does not accept new responses. |
| Anonymous Responses | When enabled, respondent emails are not stored. |
| Manage Permissions | Configure who can view the form and submitted responses. See [Manage access](#manage-access). |
| Edit Theme | Change the form's color theme. |
| Clone Form | Create a copy of the form. |
| Export Form | Download the form as a JSON file. |

For more information on managing responses, see [Form responses][3].

> ## Share a form

Reviewer 2: Missing here:
1. Adding a form (not form responses) to a dashboard: add a dashboard widget, go to the apps tab, select form widget, select your form
2. Adding a form to the self service page - click on share internally, turn on the "self service action" toggle

To configure sharing for a form:
1. From the [Forms][1] page, click a form.
1. Click {{< ui >}}Share{{< /ui >}}.

The following sharing options are available:

| Option | Description |
|--------|-------------|
| Share within Datadog | Share the form with users in your Datadog organization. |

> | Share with external users | Share the form with users outside your Datadog organization. Add recipients by email address or domain, or share with anyone with a link. You can configure an access expiration date for either option. |

Reviewer 2: This sharing externally experience might be confusing to customers so I think we should elaborate a bit more with:
1. Add an example for company domain (*@companyName.com)
2. Explain that you can create multiple sharing options with different settings and different expiration dates.
3. show an example for each option (sharable link / individuals / company domain)

To pause or remove external sharing, click {{< ui >}}Share{{< /ui >}}, then click {{< ui >}}Edit{{< /ui >}} and select {{< ui >}}Pause Sharing{{< /ui >}} or {{< ui >}}Delete Sharing{{< /ui >}}.

## Add automation

After creating a form, you can add an [action][4] or [workflow blueprint][5] that triggers automatically when a form is submitted.
1. From the [Forms][1] page, click a form.
1. At the top of the form, select {{< ui >}}Automation{{< /ui >}}.
1. Choose an action or blueprint.
1. The action or blueprint opens in a workflow canvas, where you can [edit it][6].
1. Click {{< ui >}}Create{{< /ui >}}.

> **Note**: Automations triggered by forms appear under [Workflow Automation][7]. There is no charge for workflow executions triggered by a form.

Reviewer 2: This is incorrect. Forms are free but we charge ONLY for workflow executions.

Author: Should we publicize this?

```suggestion
**Note**: Automations triggered by forms appear under [Workflow Automation][7].
```
